File: airbyte-server/src/main/java/io/airbyte/server/apis/DeploymentMetadataApiController.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.server.apis;

import io.airbyte.api.generated.DeploymentMetadataApi;
import io.airbyte.api.model.generated.DeploymentMetadataRead;
import io.airbyte.commons.server.handlers.DeploymentMetadataHandler;
import io.micronaut.context.annotation.Context;
import io.micronaut.http.annotation.Controller;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;

@Controller("/api/v1/deployment/metadata")
@Context
@Secured(SecurityRule.IS_AUTHENTICATED)
public class DeploymentMetadataApiController implements DeploymentMetadataApi {

  private final DeploymentMetadataHandler deploymentMetadataHandler;

  public DeploymentMetadataApiController(final DeploymentMetadataHandler deploymentMetadataHandler) {
    this.deploymentMetadataHandler = deploymentMetadataHandler;
  }

  @Override
  public DeploymentMetadataRead getDeploymentMetadata() {
    return ApiHelper.execute(() -> deploymentMetadataHandler.getDeploymentMetadata());
  }

}

File: airbyte-server/src/main/kotlin/io/airbyte/server/apis/controllers/DeploymentMetadataApiController.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.server.apis.controllers

import io.airbyte.api.generated.DeploymentMetadataApi
import io.airbyte.api.model.generated.DeploymentMetadataRead
import io.airbyte.commons.auth.roles.AuthRoleConstants
import io.airbyte.commons.server.handlers.DeploymentMetadataHandler
import io.airbyte.server.apis.execute
import io.micronaut.context.annotation.Context
import io.micronaut.http.annotation.Controller
import io.micronaut.security.annotation.Secured
import io.micronaut.security.rules.SecurityRule

@Controller("/api/v1/deployment/metadata")
@Context
@Secured(SecurityRule.IS_AUTHENTICATED, AuthRoleConstants.DATAPLANE)
class DeploymentMetadataApiController(
  private val deploymentMetadataHandler: DeploymentMetadataHandler,
) : DeploymentMetadataApi {
  override fun getDeploymentMetadata(): DeploymentMetadataRead? = execute { deploymentMetadataHandler.getDeploymentMetadata() }
}
